### DISCUSSION TITLE
Should IDE goto `std` source definitions

### DISCUSSION THREAD

**Discussion by kobyhallx** [noir-lang/noir#3722] [Ideas]
### Problem

Should IDE plugin and `lsp` realize requests to jump to definitions/declarations/etc of `std` library?

We deliver this library "embedded".

### Happy Case

IDE plugins understand a virtual file system (either on ide side or lsp side) and is able to open source of `std` library and navigate to location of a definition.

### Additional Context

Tracking in 
feat(lsp): goto definitions of std library #3721

### Would you like to submit a PR for this Issue?

No

**Comment by Savio-Sou** [comment]
Should users in the ultimate picture be able to goto not just `std` but all imported sources, as long as they are already stored and available on users' local machines?

**Reply by TomAFrench** [reply]
The difficulty is that the stdlib is embedded into the nargo binary so it's not as simple as "go read the file at this location and display it to the user" as it is for other libraries.

**Comment by jfecher** [comment]
I would think yes. To the best of the LSP's ability it should be able to go to any definition be it in the std or a separate crate. This can help if a user doesn't know where a definition comes from or wants to see how something is implemented. E.g. going to the definition of methods on the `Option` type to see how `and_then` differs from `map` in code if the documentation was confusing.
